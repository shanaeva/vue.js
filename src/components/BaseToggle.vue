<template>
  <div
    v-font:uppercase
    class="base-toggle"
  >
    <p class="label">
      {{ label }}
    </p>
    <button
      v-font:uppercase
      class="button aqa-button-first"
      :class="{ active: isFirstActive}"
      @click="primaryClick"
    >
      {{ buttonPrimary }}
    </button>
    <button
      v-font:uppercase
      class="button aqa-button-second"
      :class="{ active: !isFirstActive }"
      @click="secondaryClick"
    >
      {{ buttonSecondary }}
    </button>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'BaseToggle',
  props: {
    label: { type: String, required: true },
    buttonPrimary: { type: String, required: true },
    buttonSecondary: { type: String, required: true },
  },
  emits: ['firstToggleClick', 'secondToggleClick'],
  data: () => ({ isFirstActive: true }),
  methods: {
    primaryClick() {
      this.$emit('firstToggleClick');
      this.isFirstActive = true;
    },
    secondaryClick() {
      this.$emit('secondToggleClick');
      this.isFirstActive = false;
    },
  },
});
</script>

<style scoped>
  .base-toggle {
    display: flex;
    align-items: center;
    color: #ffffff;
  }
  .label {
    margin-right: 20px;
  }
  .button {
    padding: 8px 30px;
    background: #424242;
    color: #ffffff;
    border: none;
    cursor: pointer;
  }
  .active {
    background: #f65261;
  }
</style>
