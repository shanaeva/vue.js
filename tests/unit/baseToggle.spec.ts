import { shallowMount } from '@vue/test-utils';
import BaseToggle from '@/components/BaseToggle.vue';

describe('BaseToggle', () => {
  it('render', () => {
    const wrapper = shallowMount(BaseToggle, { props: { label: 'search by', buttonPrimary: 'title', buttonSecondary: 'genre' } });
    expect(wrapper.props()).toStrictEqual({ label: 'search by', buttonPrimary: 'title', buttonSecondary: 'genre' });
  });
  it('render correctly', () => {
    const wrapper = shallowMount(BaseToggle, { props: { label: 'search by', buttonPrimary: 'title', buttonSecondary: 'genre' } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
