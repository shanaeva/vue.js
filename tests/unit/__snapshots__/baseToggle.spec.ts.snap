// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseToggle render correctly 1`] = `
<div
  class="base-toggle"
>
  <p
    class="label"
  >
    search by
  </p>
  <button
    class="button aqa-button-first active"
  >
    title
  </button>
  <button
    class="button aqa-button-second"
  >
    genre
  </button>
</div>
`;
